From 5a45f71f70e073acf1b3016e851ee710eb5bdec9 Mon Sep 17 00:00:00 2001
From: Bastian Bittorf <bittorf@bluebottle.com>
Date: Sat, 16 May 2015 19:17:52 +0200
Subject: [PATCH] base-files: hotplug: remove some redundant calls and remove
 explicit subshelling

when a script fails during sourcing, the next one is taken.
this only works, if all scripts work without 'exit' calls

Signed-off-by: Bastian Bittorf <bittorf@bluebottle.com>
---
 package/base-files/files/sbin/hotplug-call |   27 +++++++++++++++------------
 1 file changed, 15 insertions(+), 12 deletions(-)

diff --git a/package/base-files/files/sbin/hotplug-call b/package/base-files/files/sbin/hotplug-call
index 260be0b..53e9bf7 100755
--- a/package/base-files/files/sbin/hotplug-call
+++ b/package/base-files/files/sbin/hotplug-call
@@ -1,18 +1,21 @@
 #!/bin/sh
-# Copyright (C) 2006-2010 OpenWrt.org
-
-export HOTPLUG_TYPE="$1"
+# Copyright (C) 2006-2015 OpenWrt.org
 
 . /lib/functions.sh
 
-PATH=/bin:/sbin:/usr/bin:/usr/sbin
-LOGNAME=root
-USER=root
-export PATH LOGNAME USER
+export HOTPLUG_TYPE="$1"
+export PATH=/bin:/sbin:/usr/bin:/usr/sbin
+export LOGNAME=root
+export USER=root
 export DEVICENAME="${DEVPATH##*/}"
 
-[ \! -z "$1" -a -d /etc/hotplug.d/$1 ] && {
-	for script in $(ls /etc/hotplug.d/$1/* 2>&-); do (
-		[ -f $script ] && . $script
-	); done
-}
+DIR="/etc/hotplug.d/$HOTPLUG_TYPE/*"
+for FILE in $DIR; do
+	case "$FILE" in
+		"$DIR")
+		;;
+		*)
+			command . $FILE
+		;;
+	esac
+done
-- 
1.7.10.4

